### PR TITLE
docs: improve `cookbook/transaction-request`

### PR DESCRIPTION
**Pull request by Torres-ssf**
# Release notes

In this release, we:

- Improve documentation for `cookbook/transaction-request`

# Summary

This PR introduces improvements to the transaction request cookbook documentation.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: .changeset/spicy-mails-sort.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+---
+
+docs: improve `cookbook/transaction-request`
```
Comment (danielbate) [inline .changeset/spicy-mails-sort.md:4]: I think this deserves to be included in release notes.

File: apps/docs/src/guide/transactions/snippets/transaction-request/fetch-coins.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const transactionRequest = new ScriptTransactionRequest({
+  script: ScriptSum.bytecode,
+});
+
+// #region transaction-request-6
+// Fetching coins
+const { coins } = await wallet.getCoins(baseAssetId);
+const { messages } = await wallet.getMessages();
+
+// Adding an specific coin or message
```
Comment (danielbate) [inline apps/docs/src/guide/transactions/snippets/transaction-request/fetch-coins.ts:19]:
```suggestion
// Adding a specific coin or message
```

File: apps/docs/src/guide/transactions/transaction-request.md [hunk 3]
```diff
@@ -30,31 +30,57 @@ A `CreateTransactionRequest` is used for create transactions, which are transact
 
 Once you have instantiated a transaction request, you can modify it by setting the transaction parameters and policies. This can either be done manually by directly altering the transaction request object, or through helper methods that are available on the above classes.
 
-### Adding Resources to a Transaction Request
+### Adding `OutputCoin`
 
-Resources populate the inputs and outputs of a transaction request. This can take the form of coins, messages or contracts. The SDK provides a range of methods for dealing with resources. Below will detail how coins and messages can be added to a transaction request.
+Including `OutputCoins` in the transaction request specifies the UTXOs that will be created once the transaction is processed. These UTXOs represent the amounts being transferred to specified account addresses during the transaction:
```
Comment (danielbate) [inline apps/docs/src/guide/transactions/transaction-request.md:35]:
```suggestion
Including `OutputCoin`s in the transaction request specifies the UTXOs that will be created once the transaction is processed. These UTXOs represent the amounts being transferred to specified account addresses during the transaction:
```

File: apps/docs/src/guide/transactions/transaction-request.md [hunk 4]
```diff
@@ -30,31 +30,57 @@ A `CreateTransactionRequest` is used for create transactions, which are transact
...
-### Adding a Contract to a Transaction Request
+### Estimating and Funding the Transaction Request
 
-Scripts can perform multiple actions on chain, therefore you may want to chain contract calls. For this you will need to add a contract to the transaction request. This can be done like so:
+Before submitting a transaction, it is essential to ensure it is properly funded to meet its requirements and cover the associated fee:
 
-<<< @./snippets/transaction-request/fund-request.ts#transaction-request-4{ts:line-numbers}
+<<< @./snippets/transaction-request/estimate-and-fund.ts#transaction-request-4{ts:line-numbers}
+
+This is the recommended approach for manually estimating and funding a transaction before submission. It ensures that the `gasLimit` and `maxFee` are accurately calculated, and that the required amounts for `OutputCoins` are fulfilled. The `fund` method automatically fetches any missing resource amounts from the calling account and adds them to the transaction request.
```
Comment (danielbate) [inline apps/docs/src/guide/transactions/transaction-request.md:45]:
```suggestion
This is the recommended approach for manually estimating and funding a transaction before submission. It ensures that the `gasLimit` and `maxFee` are accurately calculated and that the required amounts for `OutputCoin`s are fulfilled. The `fund` method automatically fetches any missing resource amounts from the calling account and adds them to the transaction request.
```

File: apps/docs/src/guide/transactions/transaction-request.md [hunk 5]
```diff
@@ -30,31 +30,57 @@ A `CreateTransactionRequest` is used for create transactions, which are transact
...
+
+<<< @./snippets/transaction-request/fetch-resources.ts#transaction-request-5{ts:line-numbers}
+
+> **Note**: While manually fetching and adding resources to a transaction request is possible, it is generally not recommended as it potentially leads to funding errors.
+
+#### Manually Fetching Coins or Messages
+
+If needed, you can manually include specific coins or messages in the transaction. However, this approach is generally discouraged and should only be used in scenarios where explicitly adding particular coins or messages to the transaction request is required:
+
+<<< @./snippets/transaction-request/fetch-coins.ts#transaction-request-6{ts:line-numbers}
```
Comment (danielbate) [inline apps/docs/src/guide/transactions/transaction-request.md:59]: The note and paragraph here seem to be saying the same thing, do we need the note?

File: apps/docs/src/guide/transactions/snippets/transaction-request/add-witness.ts [hunk 6]
```diff
@@ -16,23 +16,23 @@ const transactionRequest = new ScriptTransactionRequest({
   script: ScriptSum.bytecode,
 });
 
-// #region transaction-request-6
+// #region transaction-request-10
 // Add a witness directly
 transactionRequest.addWitness(witness);
```
Comment (petertonysmith94) [inline apps/docs/src/guide/transactions/snippets/transaction-request/add-witness.ts:21]: Is it worth showing a signature as the witness here?

Maybe generated via `account.signTransaction(transactionRequest)`

File: apps/docs/scripts/wrap-snippets.ts [hunk 7]
```diff
@@ -85,6 +85,8 @@ export const wrapSnippet = (filepath: string) => {
       .replace(/import.*$/gm, '') // ignore file imports
       .trim() // zip
       .replace(/\n/g, '\n  '); // indent chunk
+
+    console.log(nodeLauncher);
```
[the remark this answers is not in the document]
Comment (Torres-ssf) [reply apps/docs/scripts/wrap-snippets.ts:89]: 🫠

File: apps/docs/src/guide/transactions/transaction-request.md (hunk 5 above)
Comment (Torres-ssf) [reply apps/docs/src/guide/transactions/transaction-request.md]: https://github.com/FuelLabs/fuels-ts/pull/3440/commits/c67f9ae09150c576c425809a96c03c07ff20a242

File: apps/docs/src/guide/transactions/snippets/transaction-request/add-witness.ts (hunk 6 above)
Comment (Torres-ssf) [reply apps/docs/src/guide/transactions/snippets/transaction-request/add-witness.ts]: https://github.com/FuelLabs/fuels-ts/pull/3440/commits/b94fb0a0e783e8a1a467a92e89bf870cfefb9f6d

File: .changeset/spicy-mails-sort.md (hunk 1 above)
Comment (Torres-ssf) [reply .changeset/spicy-mails-sort.md:4]: @danielbate, I was unaware that we could include changes that did not affect any package in the release notes. How can I achieve this?

Comment (danielbate) [reply .changeset/spicy-mails-sort.md:4]: Just by adding release notes in PR desc I believe? 🤔

File: apps/docs/src/guide/transactions/snippets/transaction-request/add-witness.ts (hunk 6 above)
Comment (petertonysmith94) [reply apps/docs/src/guide/transactions/snippets/transaction-request/add-witness.ts]: I was thinking both side by side:
```ts
// Add a witness signature directly
const signature = account.signTransaction(transactionRequest);
transactionRequest.addWitness(witness);

// Or add multiple via `addAccountWitnesses`
await transactionRequest.addAccountWitnesses([accountA, accountB]);
```

Comment (Torres-ssf) [reply apps/docs/src/guide/transactions/snippets/transaction-request/add-witness.ts]: @petertonysmith94 seems reasonable https://github.com/FuelLabs/fuels-ts/pull/3440/commits/a847cdb43931e1691218cfa9904f1402da158e16

**Review by petertonysmith94 — APPROVED** [review]
🚀

**Review by danielbate — APPROVED** [review]

**Review by maschad — APPROVED** [review]
Great work!

**Pull request merged**